src/dependencies.js

```diff
@@ -84,33 +84,66 @@ const getFileDependencies = async function(path, packageJson, state) {
   const dependencies = precinct.paperwork(path, { includeCore: false })
 
   const depsPaths = await Promise.all(
-    dependencies.map(dependency => getImportDependencies(dependency, basedir, packageJson, state))
+    dependencies.map(dependency =>
+      getImportDependencies({ dependency, basedir, packageJson, state, treeShakingModule })
+    )
   )
   return [].concat(...depsPaths)
 }
 
-// `require()` statements can be either `require('moduleName')` or
-// `require(path)`
-const getImportDependencies = function(dependency, basedir, packageJson, state) {
-  if (LOCAL_IMPORT_REGEXP.test(dependency)) {
-    return getLocalImportDependencies(dependency, basedir, packageJson, state)
+const isTreeShakingModule = function({ dependency, treeShakingModule }) {
+  return treeShakingModule === getModuleName(dependency)
+}
+
+const getImportDependencies = function({ dependency, basedir, packageJson, state, treeShakingModule }) {
+  if (LOCAL_IMPORT_REGEXP.test(dependency) || isTreeShakingModule({ dependency, treeShakingModule })) {
+    return getTreeShakedDependencies({ dependency, basedir, packageJson, state, treeShakingModule })
+  }
+
+  if (isTreeShakable(dependency)) {
+    const module = getModuleName(dependency)
```

Comment:
We might want not to use `module` as a variable name, due to its usage in Node.js, what do you think?
Note that this should not be a problem in this particular case due to variable shadowing.
For example, `moduleName` could work instead?